modules/ObjectControl/src/objectcontrol.cpp

```diff
@@ -690,6 +714,107 @@ void ObjectControl::startScenario() {
 	}
 }
 
+void ObjectControl::resetTest() {
+	// Check if we are already resetting
+	if (this->isResetting) {
+		RCLCPP_INFO(get_logger(), "Test already resetting");
+		return;
+	}
+	RCLCPP_INFO(get_logger(), "Resetting test by offering return trajectories");
+	// Request a new return trajectory for each object
+	for (auto& id : getVehicleIDs()) {
+		this->setObjectTrajectory(id);
+	}
+	this->isResetting = true;
+}
+
+void ObjectControl::reloadScenarioTrajectories() {
+	// Check if we already have loaded the scenario trajectories
+	if (!this->isResetting) {
+		RCLCPP_INFO(get_logger(), "Scenario trajectories already loaded");
+		return;
+	}
+	RCLCPP_INFO(get_logger(), "Reloading scenario trajectories");
+	// Request the scenario trajectory for each object
+	for (auto& id : getVehicleIDs()) {
+		this->setObjectTrajectory(id);
+	}
+	this->isResetting = false;
+}
+
+void ObjectControl::updateTrajectoryGUI(uint32_t id){
+	// Update the GUI with the new trajectory in local coordinates
+	ATOS::Trajectory traj = objects.at(id)->getTrajectory();
+	this->pathPublishers.emplace(id, ROSChannels::Path::Pub(*this, id));
+	this->pathPublishers.at(id).publish(traj.toPath());
+
+	// Update the GUI with the new trajectory in global coordinates
+	GeographicPositionType origin = objects.at(id)->getOrigin();
+	std::array<double,3> llh_0 = {origin.latitude_deg, origin.longitude_deg, origin.altitude_m};
+	this->gnssPathPublishers.emplace(id, ROSChannels::GNSSPath::Pub(*this, id));
+	this->gnssPathPublishers.at(id).publish(traj.toGeoJSON(llh_0));
```

Comment:
We need to implement a simple "republisher" node that takes CartesianTrajectories as input and republishes them as GnssPaths... But It is not part of this PR.